backends/tc/ebpfCodeGen.cpp

```diff
@@ -1456,8 +1464,142 @@ void ControlBodyTranslatorPNA::processFunction(const P4::ExternFunction *functio
             builder->emitIndent();
             builder->appendLine("};");
             builder->emitIndent();
+            builder->append(
+                "bpf_p4tc_entry_update(skb, &update_params, &key, sizeof(key), act_bpf)");
+        }
+        return;
+    } else if (function->expr->method->toString() == "add_entry") {
+        auto act = findContext<IR::P4Action>();
+        BUG_CHECK(act != nullptr, "%1% add_entry extern can only be used in an action", function);
+        const IR::P4Table *t = table->table->container;
+        const IR::Expression *defaultAction = t->getDefaultAction();
+        auto defaultActionName = table->getActionNameExpression(defaultAction);
+        CHECK_NULL(defaultActionName);
+        if (defaultActionName->path->name.originalName != act->name.originalName) {
+            ::error(ErrorType::ERR_UNEXPECTED,
+                    "add_entry extern can only be applied in default action of the table.");
+        }
+        BUG_CHECK(function->expr->arguments->size() == 3,
+                  "%1%: expected 3 arguments in add_entry extern", function);
+        if (!IsTableAddOnMiss(t)) {
+            ::warning(ErrorType::WARN_MISSING,
+                      "add_entry extern can only be used in an action"
+                      " of a table with property add_on_miss equals to true.");
+        }
+        auto action = function->expr->arguments->at(0);
+        auto actionName = action->expression->to<IR::StringLiteral>()->value;
+        auto param = function->expr->arguments->at(1)->expression;
+        auto expire_time_profile_id = function->expr->arguments->at(2);
+        auto controlName = control->controlBlock->getName().originalName;
+
+        if (table) {
+            auto actID = 0;
+            for (auto a : table->actionList->actionList) {
+                auto adecl = control->program->refMap->getDeclaration(a->getPath(), true);
+                auto action = adecl->getNode()->to<IR::P4Action>();
+                if (a->toString() == actionName) {
```

Comment:
originalName can be used instead of using toString()

Reply:
addressed